### ISSUE TITLE
thumbnail in blog not showing up

### ISSUE THREAD

**Issue by kenkoonwong** [hugo-apero/hugo-apero#89]
Hi. thanks so much for this theme! it's great!

I have intermittent success with thumbnail showing up in blog section. 

/content/blog/_index.md 
show_post_thumbnail: true

thumbnails and named with "feature", and placed on the directory of the blog

url: https://www.kenkoonwong.com/blog/
github: https://github.com/kenkoonwong/everyday_is_a_school_day/tree/main/content/blog

it works fine with blogdown but for some reason when using netlify, it does not seem to like it. Thanks!

**Comment by kenkoonwong** [comment]
nevermind! it was a netlify deployment issue. I logged onto netlify and retry deployment and it worked.

**Comment by bryceikeda** [comment]
@kenkoonwong I am having the same issue, and it looks like from your first URL that the second image in the list doesn't render. If you fixed it somewhere else, did you change anything during redeployment to fix your issue? On redeployment, it seems like a random image is chosen to not render.

**Comment by mccarthy-m-g** [comment]
The behaviour is random, see also #94. I posted on the netlify forums to see if they could provide some support: https://answers.netlify.com/t/netlify-creating-random-changes-to-hugo-site-on-deploy-missing-content/79960

@kenkoonwong could you please reopen this issue for visibility?

@bryceikeda what pages are you having this issue on? I noticed it on my project page, but it sounds like you and @kenkoonwong have it with your blog listings?

**Comment by kenkoonwong** [comment]
> @kenkoonwong I am having the same issue, and it looks like from your first URL that the second image in the list doesn't render. If you fixed it somewhere else, did you change anything during redeployment to fix your issue? On redeployment, it seems like a random image is chosen to not render.

I haven’t found a solid fix. Anytime when this happens I’ll hv to redeploy and hope it fixes it

**Comment by kenkoonwong** [comment]
> @kenkoonwong I am having the same issue, and it looks like from your first URL that the second image in the list doesn't render. If you fixed it somewhere else, did you change anything during redeployment to fix your issue? On redeployment, it seems like a random image is chosen to not render.

I had to clear cache and redeploy twice to get all the pictures to load

**Comment by kenkoonwong** [comment]
> The behaviour is random, see also #94. I posted on the netlify forums to see if they could provide some support: https://answers.netlify.com/t/netlify-creating-random-changes-to-hugo-site-on-deploy-missing-content/79960
> 
> @kenkoonwong could you please reopen this issue for visibility?
> 
> @bryceikeda what pages are you having this issue on? I noticed it on my project page, but it sounds like you and @kenkoonwong have it with your blog listings?
> 
> Edit: Also, the behaviour is still affecting your site @kenkoonwong. On the current deploy this post is missing the thumbnail and author byline even though both should be displayed:
> 
> ![284DB1F4-4B16-4249-AAA3-962410DD0D01](https://user-images.githubusercontent.com/51542091/204158708-d4b62405-5c7e-438c-bd0f-17e29fad384c.jpeg)

I had to clear cache and redeploy twice to get all the pictures to load

**Comment by bryceikeda** [comment]
@mccarthy-m-g Yeah mine is on the blog page. 

@kenkoonwong I've tried doing that but I always have at least one picture that does not show up...

**Comment by mccarthy-m-g** [comment]
@bryceikeda What layout are you using for your blog page? (the `layout` param in `_index.md`) And what Hugo version?

I’ve found redeploying over and over eventually lands on a build where nothing is missing, but doing that isn’t sustainable.

A Netlify support engineer responded to me on the Netlify forum post I shared earlier, so hopefully they can help.

**Comment by bryceikeda** [comment]
@mccarthy-m-g I am currently using version 0.89.4 and the list-grid layout.

**Comment by mccarthy-m-g** [comment]
Thanks! Just some notes for reference:

## Hugo versions

I'm using version 0.89.2 and Ken is using 0.102.1. So we're having the same issue across three different Hugo versions.

## summary-grid

For Bryce and I the following If statements are randomly being evaluated as `false` on our blog and project pages respectively during the Netlify build, causing the content associated with them to not show up. This behaviour only happens if the respective `show_` params are set to `true` in `_index.md`.

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/partials/shared/summary-grid.html#L7

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/partials/shared/summary-grid.html#L31-L33

The `summary-grid` partial is called here for blog pages with the `list-grid` layout:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/blog/list-grid.html#L13

And here for project pages `list-grid` layout:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/project/list-grid.html#L13

## summary-list

For Ken the following If statements are randomly being evaluated as `false` on Netlify builds:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/partials/shared/summary.html#L7

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/partials/shared/summary.html#L35-L37

The `summary` partial is called here for blog pages with the `list-sidebar` layout:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/blog/list-sidebar.html#L8

## summary-compact

One thing I've found on my site is that this behaviour does not happen on my talks page. For talks the [`summary-compact`](https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/partials/shared/summary-compact.html) partial is used for listings, which is called here with the `list` layout:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/talk/list.html#L17

And here with the `list-sidebar` layout:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/talk/list-sidebar.html#L13

## Partials and layouts

As far as I can tell, the partials are all using the same Hugo code for If statements. So if the bug is caused by the theme and not Netlify, I don't think problem is there.

However, the layouts that call these partial range over the pages slightly differently. Blogs/projects:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/project/list-grid.html#L9-L15

Talks:

https://github.com/hugo-apero/hugo-apero/blob/17f92f68899f672868c8f1e93a5b7d26232cf11d/layouts/talk/list.html#L8-L22

My Hugo is rusty so I'd have to go back to the docs to understand this better, but there could be a weird interaction happening on Netlify when ranging over the pages that's causing the random missing content.

**Comment by bryceikeda** [comment]
@mccarthy-m-g So would Edit 2 be the current workaround until the true bug is fixed? If so, (I am not well versed in Hugo) is it as simple as setting `{{ if $root.Params.show_post_thumbnail }}` in summary.html to `{{ if true}}`?

**Comment by mccarthy-m-g** [comment]
@bryceikeda Yeah, that’s the easiest workaround at the moment. I would do it for all the if statements though. Don’t directly edit the file inside the `theme` directory, instead override the partial by creating a `layouts/partials/shared/summary-grid.html` file where the path starts from your root directory.

**Comment by bryceikeda** [comment]
@mccarthy-m-g Worked! Appreciate your help on this!

**Comment by mccarthy-m-g** [comment]
@bryceikeda Awesome!

Hopefully we can get it fixed properly soon though. Still waiting to hear back from Netlify.

**Comment by kenkoonwong** [comment]
@bryceikeda @mccarthy-m-g pardon my noobness in Hugo. What do I need to do to workaround the version I’m using? Thanks!

**Comment by mccarthy-m-g** [comment]
See this page for more info on how overriding theme files works: <https://bwaycer.github.io/hugo_tutorial.hugo/themes/customizing/>

You just need to override the appropriate theme file for your site, then replace the contents of the If statements with `true`. See the posts above for the file and If statements in question.

**Comment by bryceikeda** [comment]
> summary-grid

Hi @kenkoonwong to be more explicit (I too sometimes need clearer directions), here are the steps on what I did. 

1. Go to `themes/hugo-apero/layouts/blog` (this is the page layout I was using that was not showing pictures correctly)
2. Click list-grid.html (this is the layout that I am using)
3. Here you can see it is using `{{ partial "shared/summary-grid" . }}`
4. As stated by @mccarthy-m-g, don't directly edit the file inside the theme directory, instead override the partial by creating a `layouts/partials/shared/summary-grid.html` file where the path starts from your root directory. You can find this summary-grid.html file in the `themes/hugo-apero/layouts/partials/shared/` directory. 
5. In this new summary-grid.html file, right below where it says `<!--if show_post_thumbnail is TRUE in root-->` update it to say `{{ if true }}`

Then your website should recompile and render the pictures!

**Comment by kenkoonwong** [comment]
@bryceikeda @mccarthy-m-g Thank you so much !!! Made the change as instructed and so far working very well!!
